taxi/system/system.go

```diff
@@ -0,0 +1,79 @@
+package system
+
+import (
+	"taxi/parser"
+)
+
+// CalcTotalFare calcurates a total fare based on logs. 
+// If a taxi drives midnight, caused additional fare.
+// If a taxi drives under 10 km/h, caused additional fare.
+func CalcTotalFare(logs string) (int64, error) {
+	var midnightMinutes float64
+	var totalDistance float64
+	var totalFare int64
+	records, err := parser.Parse(logs)
+	if err != nil {
+		return 0, err
+	}
+	for i:=1; i < len(records); i++ {
+		prevMinutes := records[i-1].Time.Minutes
+		currMinutes := records[i].Time.Minutes
+		var currMidnightMinutes float64 = 0
+		if checkDrivingDuringMidnight(
+			records[i-1].Time.Hours,
+			records[i].Time.Hours,
+		){
+			currMidnightMinutes += currMinutes - prevMinutes
+		}
+		if calcVelocity(currMinutes - prevMinutes, records[i].Distance) <= float64(10) {
+			totalDistance += records[i].Distance * 1.25
```

Comment:
Is this right? What about if a car was driving from 22:00 to 32:00 at the same speed?

Reply:
It is a mistake. I will modify it.